### PR TITLE
tracing: Deprecate Opencensus (@1.30)

### PR DESCRIPTION
**Pull request by phlax**
related to #9958 

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @kyessenov

**Review by kyessenov — APPROVED** [review]
Per offline discussions.

**Review by basvanbeek — APPROVED** [review]
I'm happy we're first deprecating with a notice. Personally I feel the OpenCensus project has been archived too early as OpenTelemetry has not been great at getting to production readiness (especially metrics in Go). But I do believe their decision has made it inevitable for us to get rid of non maintained code with some sense of urgency. This strikes a good balance for people needing to switch over.

File: api/envoy/config/trace/v3/opencensus.proto [hunk 1]
```diff
@@ -48,29 +48,35 @@ message OpenCensusConfig {
   reserved 7;
```
Comment (phlax) [inline api/envoy/config/trace/v3/opencensus.proto:48]: @htuch you may know why - the one thing i didnt manage to do was deprecate the enum above (in file)

not sure why - probs i was just doing it wrong

Comment (htuch) [reply api/envoy/config/trace/v3/opencensus.proto:48]: It probably only matters that you deprecate the fields that reference it.

**Review by yanavlasov — APPROVED** [review]

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**